go/vt/vtgate/engine/update.go

```diff
@@ -0,0 +1,217 @@
+/*
+Copyright 2018 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package engine
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/youtube/vitess/go/jsonutil"
+	"github.com/youtube/vitess/go/sqltypes"
+	"github.com/youtube/vitess/go/vt/sqlannotation"
+	"github.com/youtube/vitess/go/vt/vterrors"
+	"github.com/youtube/vitess/go/vt/vtgate/vindexes"
+
+	querypb "github.com/youtube/vitess/go/vt/proto/query"
+	vtrpcpb "github.com/youtube/vitess/go/vt/proto/vtrpc"
+)
+
+var _ Primitive = (*Update)(nil)
+
+// Update represents the instructions to perform an update.
+type Update struct {
+	// Opcode is the execution opcode.
+	Opcode UpdateOpcode
+
+	// Keyspace specifies the keyspace to send the query to.
+	Keyspace *vindexes.Keyspace
+
+	// Query specifies the query to be executed.
+	Query string
+
+	// Vindex and Values specify how routing must be computed
+	Vindex vindexes.Vindex
+	Values []sqltypes.PlanValue
```

Comment:
Following the example from Insert, can we add a comment describing what the contents of of `Values` is?

Reply:
Done.